content/posts/Crash Course In ML.md

```diff
@@ -0,0 +1,191 @@
++++
+author = "Fares Ismail"
+date = "2019-12-09T10:00:00+00:00"
+tags = [
+    "Machine Learning",
+    "ML",
+]
+title = "Crash Course in Machine Learning"
++++
+
+I take article requests now... 😛
+
+The following article will be a quick overview of classical machine learning and the different types and algorithms there are.
+
+Don't expect any code implementation or a hands-on lab this is just to provide an overview of the types of machine learning algorithms and when those can be used.
+
+On a high level, we can think of a machine learning model as a black box. One that will take in some input data, do something with that data and then spit out some output data.
+
+``` text
+                    +-------------------------+
+Input Data ------>  | Machine Learning Model  | ------> Output Data
+                    +-------------------------+
+
+```
+
+This black box however needs to be trained. This is where the word `Learning` comes from.
+
+Similarly, on a very high level, Machine learning models can be broken down into two categories: ``Supervised`` and ``Unsupervised``. Those categories relate to how this model will be trained.
+
+---
+
+## Supervised Learning
+
+In supervised machine learning, We rely on a `training dataset` to teach the how to make accurate future predictions. We say that we have a labeled dataset. That is a dataset containing both the input data (features) and the output data (prediction).
```

Comment:
to teach the how. you forgot to say teach what

Reply:
what I understood here is by labeled, you mean you associate an input with an output. is this what you mean?